### DISCUSSION TITLE
Use case: Squish

### DISCUSSION THREAD

**Discussion by ColoredCarrot** [VirtualDrivers/Virtual-Display-Driver#147] [Q&A] [answered]
Hi everyone!

To start off, I don't know if this is even remotely the right use case for this project. If it isn't, I apologize, and you can tell me to sod off :)

I'm looking to run an app that requires a display (specifically, [Squish](https://doc.qt.io/squish/), a GUI testing framework) on a Windows VM managed through Hyper-V.
My search has led me from discussions with their support ("it's impossible"), through trying to run the exe via Xvfb in WSL (no luck), to, now, this place.

Quick background: Connecting to the VM via RDP (Remote Desktop Protocol) works fine; I can see Squish starting the application-under-test with its GUI and everything.
However, just running Squish through a terminal (e.g. via SSH) fails, since there's no display.

Can this driver can be used so that I don't need to connect via RDP at all, but still have a "display"?

Driver installation was successful (it's now listed right below the Microsoft Remote Display Adapter when I connect through RDP), but it doesn't seem to have any noticeable effect. Squish still fails to discover the GUI application.
I also can't check the Windows display settings; it just shows "The display settings can't be changed from a remote session."

Thanks for taking the time!

**Comment by SmilerRyan** [comment, marked as answer]
Basically, once the driver is installed you would want to interact with the "display" like a normal display, and not through the remote desktop display (try anydesk, teamviewer, parsec, etc). When using the virtual display through those types of programs instead, it interacts with it normally rather than what remote desktop does, which is making a new virtual display for your session not the same as a physical (or driver in this case) does for you.

**Comment by bp100a** [comment]
I am in a similar position. I'm working on getting Squish GUI testing running on a Windows server in CircleCI. I can install the virtual display driver and verify it's installed properly but it only lists a single display resolution, 1024x768, which is not suitable for my application.
